### DISCUSSION TITLE
supabase.auth.getUser() returns undefined on pageload

### DISCUSSION THREAD

**Discussion by jonigiuro** [supabase/supabase#11874] [Questions]
As per title, when I hard refresh the page the getUser() method returns undefined, but if I change tab and then come back it returns the logged in user correctly.

```js
<script>
  import {supabase} from "../supabase.js";
  import {user} from "../stores/authStore.js";
  import Auth from "../components/Auth.svelte";
  import Navbar from "../components/Navbar.svelte";
  import { loadTodos } from "../stores/todoStore.js";

  user.set(supabase.auth.getUser())
  console.log(user.set(supabase.auth.getUser()))

  supabase.auth.onAuthStateChange(( _, session) => {
      user.set(session?.user);
      if(session?.user){
        loadTodos();
      }
  });
  
</script>

<div class="container mx-auto my-6">        
  {#if $user}
    <Navbar/>
    <p class="text-center mb-2">Hi, {$user.email}</p>
    <slot></slot>
  {:else }
    <Auth/>
  {/if}
</div>
```

I'm following this tutorial: https://www.youtube.com/watch?v=Q0XustAh080&ab_channel=JamesQQuick

**Comment by GaryAustin1** [comment]
Not sure about your tutorial, but normally when the supabase client starts up you await getSession() for the client to get the session setup.  It normally returns null if you are not signed in or being signed in, returns session immediately if signed in and session is in storage or memory, and finally returns session after refreshing the stored session is completed by the client.  This is browser side flow.
